network.go

```diff
@@ -816,6 +835,10 @@ func (n *network) getSvcRecords() []etchosts.Record {
 	sr, _ := n.ctrlr.svcDb[n.id]
 
 	for h, ip := range sr {
+		if ep != nil && strings.HasPrefix(h, ep.Name()) {
```

Comment:
I think the HasPrefix logic needs to be a little bit more robust to take care of overlapping endpoint names.